### PR TITLE
add gopherbadgeimg tool

### PR DESCRIPTION
**Pull request by taigrr**
Wrote this in a different repo @ https://github.com/taigrr/gopherbadgeimg before I realized there was an implementation here (commented out) in the cmd folder. Made some updates to make it fit better in this repo as a cmd subpackage and ported it over here.

Code is documented with the intention that inexperienced users may learn from it, so there's quite a bit more inline comments here than I'd usually add (explaining matrix iteration loops, for example).

Binary outputs the images in both the inlined byte array format I saw in the tainigo.go file (but adds the DO NOT EDIT comment at the top that's standard for codegen) and as a base64'd string like those used in the Makefile printed to stdout.

I also output a .bin file which is just raw bytes, so that go:embed can be used (which is my preferred method since editing files with inlined byte arrays can choke certain editors on low-power machines).

There's an example of how to use go:embed as well in the test file, called out again so beginners can see how to use it.

### CODE REVIEW

**Comment by taigrr** [comment]
@conejoninja forgot to tag you in initial comment.

**Comment by taigrr** [comment]
Ok I'll pull in the stdlib flag package then and rework the entrypoint a tiny bit. Remove the positional arg requirements in favor of flags. Since I'm bringing in flags i'll probably add a mode flag for output format instead of always creating all 3 formats.

I like your "show" option, I'll do something similar I think (not inline inside of the conversion loop, it'll be its own function since speed isn't really a concern and I want it to be clear to beginners what's happening)

**Comment by taigrr** [comment]
@conejoninja @deadprogram this is ready for review.

**Comment by taigrr** [comment]
@conejoninja @deadprogram I think this is ready for review + nits

**Comment by taigrr** [comment]
@conejoninja @deadprogram this should be good to review

**Comment by conejoninja** [comment]
I checked and the -show option seems to not work properly¿? I see some ***** on the terminal but it doesn't look like the image.

**Comment by taigrr** [comment]
This looks fine to me, would you be able to give a reproducer image?

image

File: cmd/gopherbadgeimg/main.go [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+	rstr = strings.ToLower(rstr)
+	pixels := strings.Split(rstr, "x")
+	if len(pixels) != 2 {
+		return 0, 0, errors.New("invalid ratio string provided")
+	}
+	x, err := strconv.Atoi(pixels[0])
+	if err != nil {
+		return 0, 0, errors.Join(errors.New("error: could not parse x coordinate count"), err)
+	}
+	y, err := strconv.Atoi(pixels[0])
```
Comment (conejoninja) [inline cmd/gopherbadgeimg/main.go:263]: this should be 
```y, err := strconv.Atoi(pixels[1])```

if not, it takes for both (x,y) only the first part of the ratio **AAA**xBBB

File: cmd/gopherbadgeimg/main.go [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+		os.Args[0],
+		os.Args[0],
+	)
+	os.Exit(1)
+}
+
+// PrintImg prints an `*` for each marked bit
+//
+// It writes to stderr so that it doesn't conflict with the base64 output
+func PrintImg(x, y int, imgBits []byte) {
```
Comment (conejoninja) [inline cmd/gopherbadgeimg/main.go:288]: This is wrong if the image is not a square, the new line occurs at the wrong time

File: cmd/gopherbadgeimg/main.go (hunk 1 above)
Comment (taigrr) [reply cmd/gopherbadgeimg/main.go:263]: ah, right you are, typo.

File: cmd/gopherbadgeimg/main.go (hunk 2 above)
Comment (taigrr) [reply cmd/gopherbadgeimg/main.go]: Yes, I tested it with many dimensions and eyeballed it but my Atoi([0]) error made them all square

I've fixed it now, used a different method which is easier to understand, will attach images

**Comment by taigrr** [comment]
Thank you for finding those issues!

Here are screenshots of it now working with all sorts of dimension combinations.
image
image
image
image

**Comment by conejoninja** [comment]
Thank you very much, sorry I took longer than I wish to review it (personal issues), but it's a great addition. Thanks!

**Pull request merged**